src/core/Akka.Remote/Transport/AkkaProtocolTransport.cs

```diff
@@ -160,14 +161,14 @@ public override Task<bool> ManagementCommand(object message)
         /// <param name="remoteAddress">TBD</param>
         /// <param name="refuseUid">TBD</param>
         /// <returns>TBD</returns>
-        public Task<AkkaProtocolHandle> Associate(Address remoteAddress, int? refuseUid)
+        public async Task<AkkaProtocolHandle> Associate(Address remoteAddress, int? refuseUid)
         {
             // Prepare a Task and pass its completion source to the manager
             var statusPromise = new TaskCompletionSource<AssociationHandle>();
 
             manager.Tell(new AssociateUnderlyingRefuseUid(SchemeAugmenter.RemoveScheme(remoteAddress), statusPromise, refuseUid));
 
-            return statusPromise.Task.CastTask<AssociationHandle, AkkaProtocolHandle>();
+            return (AkkaProtocolHandle)await statusPromise.Task;
```

Comment:
The code is simplified here, cause CastTask generated AggregatedException of AggregatedException of.... so on the higher level there was no way to reliably handle exceptions. I believe this should be relatively safe.